app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -354,6 +327,26 @@ const useFhirResourceTreeData = (
     }
   }, [nodeId, data, dispatch]);
 
+  useEffect(() => {
+    if (attributes) {
+      const attributeNodes = attributes.map((attribute) => {
+        const elementDefinition = createElementDefinition(attribute);
+        return createElementNode(elementDefinition, {});
+      });
+      dispatch(setAttributeNodes({ attributeNodes }));
```

Comment:
Here you're dispatching each time the list of attributes refresh and you're diffing to say what node to add or remove, right? Why don't we have separate actions (add/delete) for this (to prevent the need to diff) ?

Reply:
Okay this seems like a nice alternative ! I'll look into it and update the code

Reply:
Actually, there is an edge case I am not sure this would solve easily : 

Let's say we have 2 attributes with respective paths : `Identifier[0]` and `Identifier[0].type.coding[0]`
Those attributes will be in the `attributeNodes` array in the initialization, but the second one won't be injected in the tree until the user opens the `Identifier[0].type` tree node.
Whenever the user will do so, there won't be any dispatch made because both `attributeNodes` array and `prevAttributeNodes` array will still be the same..

Reply:
I guess, from the useEffect of the hook, I'll need to :

- Check for each attribute if the corresponding node is present in the tree
   - If yes, do nothing
   - If no, check if the parent parent is present in the tree (thanks to the paths)
      - If yes -> dispatch(add...)
      - If no, do nothing